addons/osfstorage/models.py

```diff
@@ -536,3 +536,14 @@ def create_waterbutler_log(self, auth, action, metadata):
             auth=auth,
             params=params
         )
+
+    @property
```

Comment:
You'd have to confirm with @sloria, but this might be a good candidate for `cached_property` as  I imagine this'll be an expensive property, but I dunno if it has any unintended side effects...

Also, does this PR deal with the bullet point in the dev notes on this ticket "Cache the total until there is a change (file uploaded, file deleted, etc.)"?

Reply:
Somewhat unrelated, but it looks like totals for linked nodes are getting calculated and displayed on the same page as the node you're on currently -- I know this doesn't count towards the "total," but it might be un-necessarily expensive if we don't care about that total here. Doesn't seem to be too much of a problem except maybe in really really big linked nodes I'm not sure!

<img width="558" alt="screen shot 2018-06-26 at 4 41 55 pm" src="https://user-images.githubusercontent.com/801594/41938088-12a7f41a-7960-11e8-9bd1-7fe26db93b29.png">

Reply:
I didn't do any bench-marking, I just since we don't really have to recurse through anything and we're doing the summing in raw sql it should just be fast enough.

A for linked nodes the ticket says they shouldn't be included in the total, but I assume they are still appear like component nodes.